src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala

```diff
@@ -228,6 +234,64 @@ class RideHailAgent(
           .copy(currentVehicle = Vector(vehicle.id), remainingShifts = shifts.get)
       }
   }
+
+  when(PendingOfflineForCharging) {
+    case ev @ Event(
+      reply @ NotifyVehicleResourceIdleReply(_, _),
+      data
+    ) =>
+      log.debug("state(RideHailAgent.PendingOfflineForCharging.NotifyVehicleResourceIdleReply): {} tick: {}",
+        ev, _currentTick)
+      val tickOption = _currentTick
+      val currentTriggerIdOption = _currentTriggerId
+      handleNotifyVehicleResourceIdleReply(reply, data)
+      requestParkingStall()
+      val tick = _currentTick.getOrElse({
+        log.warning("Current tick is empty when moving vehicle to OfflineForCharging. " +
+          "This likely will get the agent stuck")
+        0
+      })
+      val triggerId = currentTriggerIdOption.getOrElse({
+        log.warning("Current trigger id is empty when moving vehicle to OfflineForCharging. " +
+          "This COULD get the agent stuck")
+        0L
+      })
+      holdTickAndTriggerId(tick, triggerId)
+      goto(OfflineForCharging)
+  }
+
+  when(OfflineForCharging) {
+    case ev @ Event(ParkingInquiryResponse(stall, _), _) =>
+      log.debug("state(RideHailAgent.OfflineForCharging.ParkingInquiryResponse): {}", ev)
+      vehicle.useParkingStall(stall)
```

Comment:
I still have to run tests, but can anybody verify that this would be enough to move the vehicle to this location? Or do I need to first create a routing request and move the vehicle. IIRC the manager will handle that - but it doesn't hurt to verify with those who might know ;)